File: task-03/sub-task02/C/readandwrite.c
#include <stdio.h>
#include <stdlib.h>

int main()
{
    FILE *inputFile = fopen("input.txt", "r");
    if (inputFile == NULL)
    {
        printf("Input file not found.\n");
        return 1;
    }

    char inputString[100];
    fgets(inputString, 100, inputFile);
    fclose(inputFile);

    FILE *outputFile = fopen("output.txt", "w");
    if (outputFile == NULL)
    {
        printf("Error writing to output file.\n");
        return 1;
    }

    fputs(inputString, outputFile);
    fclose(outputFile);

    outputFile = fopen("output.txt", "r");
    char outputContent[100];
    fgets(outputContent, 100, outputFile);
    fclose(outputFile);

    printf("Contents of output.txt:\n%s", outputContent);

    return 0;
}

File: task-03/sub-task02/Java/readandwrite.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.FileReader;
import java.io.FileWriter;
import java.io.IOException;

public class readandwrite {
    public static void main(String[] args) {
        //filepath
        String inputFilePath = "input.txt";
        String outputFilePath = "output.txt";
         //reding
        StringBuilder content = new StringBuilder();
        try (BufferedReader inputFile = new BufferedReader(new FileReader(inputFilePath))) {
            String line;
            while ((line = inputFile.readLine()) != null) {
                content.append(line).append(System.lineSeparator());
            }
        } catch (IOException e) {
            System.out.println("Input file not found.");
            e.printStackTrace();
            return;
        }
        //using try 
        try (BufferedWriter outputFile = new BufferedWriter(new FileWriter(outputFilePath))) {
            outputFile.write(content.toString());
        } catch (IOException e) {
            System.out.println("Error writing to output file.");
            e.printStackTrace();
            return;
        }

        // Read output file
        try (BufferedReader outputFile = new BufferedReader(new FileReader(outputFilePath))) {
            String line;
            System.out.println("Contents of output.txt:");
            while ((line = outputFile.readLine()) != null) {
                System.out.println(line);
            }
        } catch (IOException e) {
            System.out.println("Error reading from output file.");
            e.printStackTrace();
        }
    }
}
